pyclam/criterion.py

```diff
@@ -337,3 +337,36 @@ def _select(root: Cluster, upper: float, lower: float, grace_depth: int) -> Set[
             inactive = {child for cluster in inactive for child in cluster.children}
 
         return selected
+
+
+class Labels(SelectionCriterion):
+    def __init__(self, labels: Dict[int, int]):
+        """ Uses a given dictionary of index -> label to select optimal clusters.
+
+        During a BFT of the tree, when a cluster has an overwhelming majority of points with the same label,
+        that cluster is selected.
+        If some data points are unlabelled, then when a cluster contains no labelled points, it too is selected.
+        """
```

Comment:
We don't usually document in `__init__`, usually docstrings go directly on the class object

Reply:
moved